### PR TITLE
Tests for events

### PR DESCRIPTION
**Pull request by sinitcin**

### CODE REVIEW

File: consensus/src/detail/test.rs [hunk 1]
```diff
@@ -1279,3 +1279,213 @@ fn test_spend_inputs_simple() {
...
+        let expected_block_height = BlockHeight::new(1);
+
+        // Event handler
+        let subscribe_func = Arc::new(
+            move |consensus_event: ConsensusEvent| match consensus_event {
+                ConsensusEvent::NewTip(block_id, block_height) => {
+                    assert!(block_height == expected_block_height);
+                    assert!(block_id == expected_block_id);
+                    // All checks went fine, let's finish the test with "no error" exit code
+                    std::process::exit(0);
```
Comment (TheQuantumPhysicist) [inline consensus/src/detail/test.rs:1309]: Wait, what?

File: consensus/src/detail/test.rs [hunk 2]
```diff
@@ -1279,3 +1279,213 @@ fn test_spend_inputs_simple() {
...
+            move |consensus_event: ConsensusEvent| match consensus_event {
+                ConsensusEvent::NewTip(block_id, block_height) => {
+                    let counter = &mut *event_counter.lock().unwrap();
+                    assert!(*counter <= COUNT_SUBSCRIBERS);
+                    assert!(block_height == expected_block_height);
+                    assert!(block_id == expected_block_id);
+                    // All checks went fine, let's finish the test with "no error" exit code
+
+                    if *counter == COUNT_SUBSCRIBERS {
+                        std::process::exit(0);
```
Comment (TheQuantumPhysicist) [inline consensus/src/detail/test.rs:1361]: I must be missing something... why are we manually existing a *process*?

File: consensus/src/detail/test.rs [hunk 3]
```diff
@@ -1279,3 +1271,155 @@ fn test_spend_inputs_simple() {
...
+
+        for block in blocks {
+            // We should connect a new block
+            assert!(consensus.process_block(block.clone(), BlockSource::Local).is_ok());
+        }
+    });
+}
+
+#[test]
+fn test_events_fail_block() {
```
Comment (TheQuantumPhysicist) [inline consensus/src/detail/test.rs:1396]: test_events_orphan_block()

File: consensus/src/detail/test.rs (hunk 1 above)
Comment (sinitcin) [reply consensus/src/detail/test.rs]: Removed

File: consensus/src/detail/test.rs (hunk 2 above)
Comment (sinitcin) [reply consensus/src/detail/test.rs]: Removed, my bad

File: consensus/src/detail/test.rs (hunk 3 above)
Comment (sinitcin) [reply consensus/src/detail/test.rs]: Done!

**Review by TheQuantumPhysicist — APPROVED** [review]

**Pull request merged**